### ISSUE TITLE
Feature Request: Consider adding a function to retrieve cardinality of a bucket

### ISSUE THREAD

**Issue by rickspencer3** [influxdata/flux#3122]
Cardinality is an important aspect of schema design that impacts both performance and limits. It is possible to run a query to calculate this, but that query is expensive and inscrutable. 

I would suggest something like 

```
cardinality(bucket: my_bucket)
```

The push the whole operation down to the storage tier so it is reasonably fast.

**Comment by nathanielc** [comment]
As a first pass I believe this can be implemented as:

```
cardinality = (bucket) =>
    from(bucket:bucket) 
       |> range()
       |> filter // select the time range, field and measurement that you care about
       |> last()
       |> count()
       |> group()
       |> sum()
```


Currently the `from |> range |> filter |> last` will be pushed down to storage but not the rest of it. 

We should benchmark the query to see how we can make it faster.

**Comment by jacobmarble** [comment]
if `|> cardinality()` were a flux function, then I think the storage engine could implement it as `from |> range |> filter |> cardinality`.

**Comment by rickspencer3** [comment]
Didn't Jacob say that related functionality is already built into the storage engine. Would it make sense/be possible to consider exposing that to Flux?

**Comment by jlapacik** [comment]
@nathanielc that query times out for high cardinality data so I think we'll definitely need to take advantage of any storage engine metrics that might be available.

**Comment by nathanielc** [comment]
In summary we know of three ways we could accomplish this request:

1. Do a scan of the TSM files and optmize that in the storage engine (this is the long query I posted above)
2. Do a scan of the TSM index files, do the storage meta APIs already allow for this? Or would we need to add another RPC call?
3. Make an API call to get the cached cardinality for a bucket that the storage engine uses to enforce cardinality limits.

We prefer 3 as it seems to be the fastest way to retrieved the data since it won't need a scan. We will sync up with the storage team to learn how that could work.

**Comment by jacobmarble** [comment]
(2) would be the most flexible. The indexes contain this info, and they are cached and fast for queries just like this, scoped to a given predicate like bucket/measurement/tag. We would need a new API method for this.

**Comment by jacobmarble** [comment]
For details on how to implement (2), see the implementations of `TagValues`, `MeasurementNames`, etc.

**Comment by nathanielc** [comment]
See for all the bits to implement this:

https://github.com/influxdata/idpe/issues/8428 
https://github.com/influxdata/idpe/issues/8430
https://github.com/influxdata/idpe/issues/8432
https://github.com/influxdata/idpe/issues/8433
https://github.com/influxdata/idpe/issues/8322

**Comment by philjb** [comment]
This is fully implemented now, right?

**Comment by tara-influxdata** [comment]
@rickspencer3 Has this been implemented? Is it available in any release? :D Thank you!

**Comment by sanderson** [comment]
@tara-influxdata `The influxdb.cardinality()` function was released in Flux 0.92.0, but it only works with Cloud. The API this function uses to retrieve the cardinality hasn't been implemented in OSS yet (as far as I know). @jlapacik?
